### PR TITLE
Feature create ui forgot the password

### PR DESCRIPTION
**Pull request by nguyenhuy95dn**

### CODE REVIEW

File: app/src/main/java/com/framgia/fdms/screen/forgetpassword/ForgetpasswordActivity.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import android.databinding.DataBindingUtil;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import com.framgia.fdms.R;
+import com.framgia.fdms.databinding.ActivityForgetpasswordBinding;
+
+/**
+ * Forgetpassword Screen.
+ */
+public class ForgetpasswordActivity extends AppCompatActivity {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fdms/screen/forgetpassword/ForgetpasswordActivity.java]: should rename ForgetpasswordActivity > ForgotPasswordActivity
and package to forgotpassword

File: app/src/main/java/com/framgia/fdms/screen/forgetpassword/ForgetpasswordPresenter.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Override
+    public void onStop() {
+        //TODO dev later
+    }
+
+    @Override
+    public boolean validateDataInput() {
+        boolean isValid = true;
+        if (TextUtils.isEmpty(mViewModel.getEmail())) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fdms/screen/forgetpassword/ForgetpasswordPresenter.java]: should write 
 if (TextUtils.isEmpty(mViewModel.getEmail())) {
            mViewModel.onInputEmailError();
               return false;
        }
        return true;

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -37,6 +37,10 @@
...
             android:name=".screen.search.SearchActivity"
             android:screenOrientation="portrait"/>
+        <activity
+            android:name=".screen.forgotpassword.ForgotpasswordActivity"
+            android:screenOrientation="portrait"
+            android:theme="@style/Theme.Design.NoActionBar"/>
     </application>
 
-</manifest>
+</manifest>
```
Comment (daolq3012) [inline app/src/main/AndroidManifest.xml]: (2tat)

File: app/src/main/java/com/framgia/fdms/screen/forgotpassword/ForgetpasswordContract.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * This specifies the contract between the view and the presenter.
+ */
+interface ForgetpasswordContract {
+    /**
+     * View.
+     */
+    interface ViewModel extends BaseViewModel<Presenter> {
+        void onInputEmailError();
+
+        String getEmail();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fdms/screen/forgotpassword/ForgetpasswordContract.java]: ?

File: app/src/main/java/com/framgia/fdms/screen/forgotpassword/ForgetpasswordPresenter.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Override
+    public void onStop() {
+        //TODO dev later
+    }
+
+    @Override
+    public boolean validateDataInput() {
+        if (TextUtils.isEmpty(mViewModel.getEmail())) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fdms/screen/forgotpassword/ForgetpasswordPresenter.java]: email must math with email format (research patern email java)

File: app/src/main/java/com/framgia/fdms/screen/forgotpassword/ForgetpasswordContract.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * This specifies the contract between the view and the presenter.
+ */
+interface ForgetpasswordContract {
+    /**
+     * View.
+     */
+    interface ViewModel extends BaseViewModel<Presenter> {
+        void onInputEmailError();
+        void onInputFormatEmailError();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fdms/screen/forgotpassword/ForgetpasswordContract.java]: please add 1 empty line between 2 method.

File: app/src/main/java/com/framgia/fdms/screen/forgotpassword/ForgetpasswordPresenter.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Listens to user actions from the UI ({@link ForgotpasswordActivity}), retrieves the data and
+ * updates
+ * the UI as required.
+ */
+final class ForgetpasswordPresenter implements ForgetpasswordContract.Presenter {
+    private static final String TAG = ForgetpasswordPresenter.class.getName();
+
+    private final ForgetpasswordContract.ViewModel mViewModel;
+    private String mRegex =
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fdms/screen/forgotpassword/ForgetpasswordPresenter.java]: change to constant variable like
``private static final String EMAIL_REGEX = ...``

File: app/src/main/java/com/framgia/fdms/screen/main/MainActivity.java [hunk 8]
```diff
@@ -19,6 +20,10 @@
 public class MainActivity extends AppCompatActivity {
     private MainContract.ViewModel mViewModel;
 
+    public static Intent getMainIntent(Context context) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fdms/screen/main/MainActivity.java]: Please change to``getInstance`` look like better

File: app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -37,6 +37,11 @@
         <activity
             android:name=".screen.search.SearchActivity"
             android:screenOrientation="portrait"/>
+        <activity
+            android:name=".screen.forgotpassword.ForgotpasswordActivity"
+            android:screenOrientation="portrait"
+            android:theme="@style/Theme.Design.NoActionBar"/>
```
Comment (daolq3012) [inline app/src/main/AndroidManifest.xml]: remove this line, It unnecessary because theme ``NoActionBar`` declared in app

**Comment by daolq3012** [comment]
LGTM!

**Pull request merged**